src/nccl_ofi_rdma.c

```diff
@@ -3553,12 +3457,12 @@ static nccl_net_ofi_rdma_recv_comm_t *prepare_recv_comm(nccl_net_ofi_rdma_listen
 	r_comm->base.close = recv_close;
 
 	/* Allocate recv communicator ID */
-	int comm_id = nccl_ofi_idpool_allocate_id(ep->comm_idpool);
-	if (OFI_UNLIKELY(comm_id < 0)) {
+	uint32_t comm_id = (uint32_t)nccl_ofi_idpool_allocate_id(ep->comm_idpool);
+	if (OFI_UNLIKELY(((int)comm_id) < 0)) {
```

Comment:
You still need to check for negative return values from `nccl_ofi_idpool_allocate_id`. Then you can cast to `uint32_t` after.

Reply:
I am casting to `int` to check the return value to avoid having to allocate 2 different variables, but I guess I can just make it `int`.

Reply:
The point is, if you immediately cast the return of `nccl_ofi_idpool_allocate_id` to uint32_t, you'll lose any negative return value from that function.

Reply:
well, you don't lose any negative return value from the function.  `(int)-1` is equal to `(int)(unsigned int)(int)-1)` on any 2s compliment architecture, but it's also really bad practice.  So you're both wrong, but Eric's right that you should just use two variables, and let the compiler collapse that into a single register (which it almost surely will do).

Reply:
Right, as long as `int` is 32 bits I am not losing the negative value and the code that I had worked correctly, but I see it that it is ugly. I changed to `int`, since I am immediately reassigning it anyway.